[Docs] Prepare docs for refactoring (#678)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md

# Model usage

> This guide shows you how to setup a LLM model to be used for Vizro-AI. Setting up a LLM is required for the package to generate charts and dashboards based on natural language queries.

Reviewer: ```suggestion
This guide shows how to set up a large language model (LLM) for use with Vizro-AI. Setting up a LLM is required for the package to generate charts and dashboards based on natural language queries.
```

```py
vizro_ai = VizroAI(model="<chosen model>")
```

!!!note
    For the string settings to work, you must provide your API key via environment variables. The relevant variable names to be set are noted in each vendor tab.

=== "OpenAI"

    | Env variable      | Name(s)                               |
    | -----------       | ------------------------------------  |
    | API key           | `OPENAI_API_KEY`                      |
    | Base API URL      | `OPENAI_API_BASE`                     |

    To use OpenAI with Vizro-AI, you must have an account with paid-for credits available. None of the free accounts will suffice. You can check [all available OpenAI models including pricing on their website](https://platform.openai.com/docs/models). This will also explain which version the below string acronyms currently point to.

    - `gpt-4o-mini` **default**
    - `gpt-4-turbo`
    - `gpt-4o`
    - `gpt-4`
    - `gpt-3.5-turbo`

> === "Anthropic"

Reviewer 2: Other vendors are not working properly with dashboard yet. We shall note these only works for chart creation now, not for dashboard

### Setting model via class for additional configuration
Beyond passing a string, you can pass **any** model derived from [Langchains `BaseChatModel` class](https://api.python.langchain.com/en/latest/language_models/langchain_core.language_models.chat_models.BaseChatModel.html#langchain_core.language_models.chat_models.BaseChatModel) that has the `with_structured_output` method implemented. An overview of the [most common vendor models supporting this functionality](https://python.langchain.com/v0.2/docs/integrations/chat/) can be found in Langchains documentation.

When choosing this approach, you can customize your model beyond the chosen default from the string instantiation. The choice of available arguments depends on the specific vendor implementation, but usually the main parameter to tweak is the temperature.

<!-- vale off -->
To ensure a deterministic answer to our queries, we've set the temperature to 0 in the string instantiation. If you prefer more creative (but potentially more unstable) responses, you can raise the temperature to a maximum of 1.
<!-- vale on -->

> Below you can find an example where a custom model is instantiated with various custom parameters. Note that we manually set the API key and base URL, which is sometimes the easiest way to get it setup.

Reviewer: ```suggestion
Below you can find an example where a custom model is instantiated with various custom parameters. Note that we manually set the API key and base URL, which is the easiest way to get it set up.
```

```py
import os
from langchain_openai import AzureChatOpenAI

llm = AzureChatOpenAI(
    azure_deployment="gpt-4-1106-preview",
    api_version="2024-04-01-preview",
    temperature=0.4,
    azure_endpoint=os.environ["AZURE_OPENAI_ENDPOINT"],
    api_key=os.environ["AZURE_OPENAI_API_KEY"]
)

vizro_ai = VizroAI(model=llm)
```

Passing an instantiated model to `VizroAI` lets you customize it, and additionally, it enables you to use a model or vendor that is not included in the above list of [string model names](#setting-model-via-string-for-ease-of-use).

## What model to choose?

### Chart generation

> At the time of writing, we found that for chart creation, some of the leading vendor's "cheaper" models, for example OpenAI's `gpt-4o-mini` and `gpt-3.5` model series, have lower price points and faster speeds, and are sufficient for basic charting.

Reviewer: lower than what and faster than what? I think maybe a slight revision, see suggestion but feel free to modify...

> At the time of writing, we found that for chart creation, some of the leading vendor's "cheaper" models, for example OpenAI's `gpt-4o-mini` and `gpt-3.5` model series, have lower price points and faster speeds, and are sufficient for basic charting.

Reviewer: I wouldn't mention speed because it seems like the faster the better, but you're implying the opposite here.
